### DISCUSSION TITLE
SVG engine

### DISCUSSION THREAD

**Discussion by yanjustino** [8T4/c4sharp#2] [Ideas]
Well, Plantuml is cool but... 

I've thinking create a new C4 Engine to replace plantuml. My ideia is use SVG over html. For this, we should change the way that Structure class parse contents in `ToString()` method. Probably we will need of strategy pattern to switch between PLANTUML and SVG results.

**Comment by jtreher** [comment]
I think this is an interesting idea. The most challenge aspect seems to be how you'd auto-balance the diagram. PlantUML doesn't always do a great job with this, but at least it avoids too many overlapping lines most of the time.
